### PR TITLE
refactor(query): rewrite avg function and fold count

### PR DESCRIPTION
**Pull request by zhang2014**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

improve(query): rewrite avg function and fold count

- Closes #issue

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/13064)

### CODE REVIEW

**Review by sundy-li — APPROVED** [review]

File: src/query/sql/src/planner/semantic/aggregate_rewriter.rs [hunk 1]
```diff
@@ -26,47 +26,74 @@ pub struct AggregateRewriter {
...
-                    Expr::BinaryOp {
-                        op, left, right, ..
-                    } if matches!(op, BinaryOperator::Minus | BinaryOperator::Plus) => {
-                        match (left.as_ref(), right.as_ref()) {
-                            (l @ Expr::Literal { .. }, other) => {
-                                let text = format!("{} * count() {op} sum({})", l, other);
+            } if !*distinct && args.len() == 1 && window.is_none() => {
+                match name.name.to_ascii_lowercase().to_lowercase().as_str() {
+                    "sum" => self.rewrite_sum(args),
+                    "avg" => self.rewrite_avg(args),
```
Comment (sundy-li) [inline src/query/sql/src/planner/semantic/aggregate_rewriter.rs:39]: maybe we should rewrite `avg(arg)` only if the there is `sum(arg)` aggregator.

**Comment by Dousir9** [comment]
We need to add some tests.

**Comment by zhang2014** [comment]
`Dev / linux / sqllogic_standalone_tpcds_native (pull_request) Failing after 1m` https://github.com/datafuselabs/databend/issues/13108

**Comment by BohuTANG** [comment]
For TPCH Q1:
main: 11.4s
this:  10.6s

**Pull request merged**